detection-rules/impersonation_mailgun.yml

```diff
@@ -0,0 +1,54 @@
+name: "Brand Impersonation: Mailgun"
+description: "Impersonation of the Mailgun Email delivery platform."
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  and 2 of (
+    regex.icontains(sender.display_name, '\bmailgun\b'),
+    strings.ilike(sender.email.domain.domain, '*mailgun*'),
+    strings.ilike(sender.email.local_part, '*mailgun*'),
+    strings.ilike(subject.subject, '*mailgun*'),
+    (
+      strings.icontains(body.current_thread.text, 'mailgun')
+      // Negate Mailgun SLDs in Microsoft Unknown Sender Warnings
+      and not (
+        strings.contains(body.current_thread.text,
+                         'https://aka.ms/LearnAboutSenderIdentification'
+        )
+        and not sender.email.email in (body.current_thread.text)
+      )
+    )
+  )
+  and (
+    any(ml.nlu_classifier(body.current_thread.text).tags,
+        .name in ("payment", "invoice")
+    )
+    or any(ml.nlu_classifier(body.current_thread.text).intents,
+           .name in ("cred_theft", "steal_pii") and .confidence == "high"
+    )
+    // Look for fake Mailgun Newsletters
+    or regex.icontains(body.current_thread.text,
+                       'unsubscribe',
+                       'trial',
+                       'blog',
+                       'plan',
+                       'upgrade',
+                       'list',
+                       'subscription'
+    )
+  )
+  and not (
+    sender.email.domain.root_domain not in ('mailgun.com', 'mailgun.net')
```

Comment:
`and not (` with `not in`

I think this `not in` should be `in` and let the outer `not` handle it